src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinInetProxyHelper.cs

```diff
@@ -178,5 +181,42 @@ public bool ManualSettingsOnly
 
             return useProxy;
         }
+
+        public bool AutoDetectScriptSuccess()
+        {
+            // Only check WinHttpDetectAutoProxyConfigUrl if auto detect is set.
+            if (!AutoDetect) return false;
+
+            bool doAutoDetect = false;
+            int currentTimeStamp = Environment.TickCount;
+
+            // Environment.TickCount will increment from zero to Int32.MaxValue for approximately 24.9 days, then jump to
+            // Int32.MinValue. When that happens, we may skip calling WinHttpDetectAutoProxyConfigUrl for one request which we
+            // otherwise will call. Since _previousTimeStampTicks will get updated in that request, subsequent requests in
+            // next 24.9 days period will behave correctly.
+            doAutoDetect = currentTimeStamp - _previousTimeStampTicks > TimerPeriodInMilliseconds;
+            _previousTimeStampTicks = currentTimeStamp;
+
+            if (doAutoDetect)
+            {
+                WinHttpTraceHelper.Trace("WinInetProxyHelper.AutoDetectScriptSuccess: Start auto discovery.");
+                IntPtr autoProxyUrl = new IntPtr();
+
+                try
```

Comment:
Is this 'try'/'finally' required here?  The WinHttp* p/invokes don't throw, so it is probably cheaper not to bother with doing try/finally at all.